content/blog/2022-08-whats-new-in-axum-0-6-0-rc1.md

```diff
@@ -0,0 +1,276 @@
+};
+use axum_extra::extract::cookie::{CookieJar, Cookie};
+
+async fn my_middleware<B>(
+    // Run the `CookieJar` extractor as part of this middleware
+    cookie_jar: CookieJar,
+    request: Request<B>,
+    next: Next<B>,
+) -> Response {
+    let response = next.run(request).await;
+
+    // Add a cookie to the jar
+    let updated_cookie_jar = cookie_jar.add(Cookie::new("foo", "bar"));
+
+    // Add the new cookies to the response
+    (updated_cookie_jar, response).into_response()
+}
+
+let app = Router::new()
+    .route("/", get(|| async { "Hello, World!" }))
+    .layer(middleware::from_fn(my_middleware));
+```
+
+Note that this cannot be used to extract `State`. See [the docs][extract state
+in middleware] for more details.
+
+## Nested routers with fallbacks
+
+`Router::nest` allows you to send all requests with a matching prefix to some
+other router or service. However in 0.5 it wasn't possible for nested routers to
+have their own fallbacks. In 0.6 that now works:
+
+```rust
+use axum::{Router, Json, http::StatusCode, routing::get};
+use serde_json::{Value, json};
+
+let api = Router::new()
+    .route("/users/:id", get(|| async {}))
+    // We'd like our API fallback to return JSON
+    .fallback(api_fallback);
+
+let app = Router::new()
+    .nest("/api", api)
+    // And our top level fallback to return plain text
+    .fallback(top_level_fallback);
+
+async fn api_fallback() -> (StatusCode, Json<Value>) {
+    let body = json!({
+        "status": 404,
+        "message": "Not Found",
+    });
+    (StatusCode::NOT_FOUND, Json(body))
+}
+
+async fn top_level_fallback() -> (StatusCode, &'static str) {
+    (StatusCode::NOT_FOUND, "Not Found")
+}
+```
+
+## Trailing slash redirects removed
+
+Previously if you had a route for `/foo` but got a request with `/foo/`, axum
+would send a redirect response to `/foo`. However many found this behavior
+surprising and it had edge-case bugs when combined with services that did their
+own redirection, so in 0.6 we decided to remove this feature.
+
+The recommended solution is to explicitly add the routes you want:
+
+```rust
+use axum::{
+    Router,
+    routing::get,
+};
+
+let app = Router::new()
+    // Send `/foo` and `/foo/` to the same handler
+    .route("/foo", get(foo))
+    .route("/foo/", get(foo));
+
+async fn foo() {}
+```
+
+If you want to opt into the old behavior you can use [`RouterExt::route_with_tsr`]
+
+## Mix wildcard routes and regular routes
+
+axum's `Router` now has better support for mixing wilcard routes and regular
+routes:
+
+```rust
+use axum::{Router, routing::get};
+
+let app = Router::new()
+    // In 0.5 these routes would be considered overlapping and not be allowed
+    // but in 0.6 it just works
+    .route("/foo/*rest", get(|| async {}))
+    .route("/foo/bar", get(|| async {}));
+```
+
+## See the changelog for more
+
+I encourage you to read the [changelog] to see all the changes and for tips on
+how to update from 0.5 to 0.6.0-rc.1.
```

Comment:
Is there a timelines for `0.6`?

Reply:
Nope. I don't wanna set any expectations I might not be able to meet. But my estimate is two weeks, depending on how many issues people find.